gasprice/gasprice.go

```diff
@@ -0,0 +1,268 @@
+package gasprice
+
+import (
+	"fmt"
+	"math/big"
+	"sort"
+	"sync"
+
+	"github.com/0xPolygon/polygon-edge/chain"
+	"github.com/0xPolygon/polygon-edge/crypto"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/umbracle/ethgo"
+)
+
+const couldNotFoundBlockFormat = "could not find block. Number: %d, Hash: %s"
+
+// DefaultGasHelperConfig is the default config for gas helper (as per ethereum)
+var DefaultGasHelperConfig = &Config{
+	NumOfBlocksToCheck: 20,
+	PricePercentile:    60,
+	SampleNumber:       3,
+	MaxPrice:           ethgo.Gwei(500),
+	LastPrice:          ethgo.Gwei(1),
+	IgnorePrice:        big.NewInt(2), // 2 wei
+}
+
+// Config is a struct that holds configuration of GasHelper
+type Config struct {
+	// NumOfBlocksToCheck is the number of blocks to sample
+	NumOfBlocksToCheck uint64
+	// PricePercentile is the sample percentile of transactions in a block
+	PricePercentile uint64
+	// SampleNumber is number of transactions sampled in a block
+	SampleNumber uint64
+	// MaxPrice is the tip max price
+	MaxPrice *big.Int
+	// LastPrice is the last price returned for maxPriorityFeePerGas
+	// when starting node it will be some default value
+	LastPrice *big.Int
+	// IgnorePrice is the lowest price to take into consideration
+	// when collecting transactions
+	IgnorePrice *big.Int
+}
+
+// Backend is the interface representing blockchain and consensus backend
+type Backend interface {
+	GetBlockByHash(hash types.Hash, full bool) (*types.Block, bool)
+	Header() *types.Header
+	Config() *chain.Params
+}
```

Comment:
Should the Backend interface be defined here?

Reply:
Since I needes only these three functions from the blockchain, I didn't want to include some interfaces we already had in polybft. Plus, this makes mocking easier since I dont need to have mock implementations of all the functions in blockchain.

Reply:
Fix: https://github.com/0xPolygon/polygon-edge/pull/1629/commits/67ff422876baedb3ccee43e6e121f07428a028f2